test/unit/org/apache/cassandra/index/sai/cql/DropIndexWhileQueryingTest.java

```diff
@@ -0,0 +1,104 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.cql;
+
+import org.apache.cassandra.index.sai.SAITester;
+import org.apache.cassandra.index.sai.plan.QueryController;
+import org.apache.cassandra.inject.ActionBuilder;
+import org.apache.cassandra.inject.Injection;
+import org.apache.cassandra.inject.Injections;
+
+import org.junit.Test;
+
+import static org.apache.cassandra.inject.InvokePointBuilder.newInvokePoint;
+import static org.junit.Assert.assertTrue;
+
+public class DropIndexWhileQueryingTest extends SAITester
+{
+    @Test
+    public void testReproductionForCNDB10732() throws Throwable
+    {
+        createTable("CREATE TABLE %s (electronicVehicle TEXT PRIMARY KEY, modelYear INT, make TEXT, vehType TEXT) WITH compaction = " +
+                "{'class' : 'SizeTieredCompactionStrategy', 'enabled' : false }");
+
+        createIndex("CREATE CUSTOM INDEX ON %s(make) USING 'StorageAttachedIndex'");
+        String indexName = createIndex("CREATE CUSTOM INDEX ON %s(modelYear) USING 'StorageAttachedIndex'");
+        createIndex("CREATE CUSTOM INDEX ON %s(vehType) USING 'StorageAttachedIndex'");
+        waitForTableIndexesQueryable();
+
+        Injection injection = Injections.newCustom("drop_index")
+                                        .add(newInvokePoint()
+                                             .onClass(QueryController.class)
+                                             .onMethod("buildPlan")
+                                             .atEntry())
+                                        .add(ActionBuilder
+                                             .newActionBuilder()
+                                             .actions()
+                                             .doAction("org.apache.cassandra.index.sai.cql.DropIndexWhileQueryingTest" +
+                                                       ".dropIndexForBytemanInjections" +
+                                                       "(\"DROP INDEX cql_test_keyspace." + indexName + "\")"))
+                                        .build();
+
+        Injections.inject(injection);
+        injection.enable();
+        assertTrue("Injection should be enabled", injection.isEnabled());
```

Comment:
The two test methods have a bunch of common code, we might want use a shared utility method:
```
private static void injectIndexDrop(String injectionName, String indexName, boolean atEntry) throws Throwable
{
    InvokePointBuilder invokePoint = newInvokePoint().onClass(QueryController.class).onMethod("buildPlan");
    Injection injection = Injections.newCustom(injectionName)
                                    .add(atEntry ? invokePoint.atEntry() : invokePoint.atExit())
                                    .add(ActionBuilder
                                         .newActionBuilder()
                                         .actions()
                                         .doAction("org.apache.cassandra.index.sai.cql.DropIndexWhileQueryingTest" +
                                                   ".dropIndexForBytemanInjections(\"" + indexName + "\")"))
                                    .build();
    Injections.inject(injection);
    injection.enable();
    assertTrue("Injection should be enabled", injection.isEnabled());
}
```